server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/KeywordsConsistencyOnDelegationMailbox.feature

```diff
@@ -0,0 +1,152 @@
+
+  Scenario: getMessageList filtered by flag should combine flag when delegation mailbox
+    Given "bob@domain.tld" set flags on "m1" to "$Flagged"
+    When "alice@domain.tld" asks for message list in mailboxes "shared,notShared" with flag "$Flagged"
+    Then the message list contains "m1"
+
+  Scenario: getMessageList filtered by flag should keep flag on delegation mailbox
+    Given "bob@domain.tld" set flags on "m1" to "$Flagged"
+    When "alice@domain.tld" asks for message list in mailboxes "notShared" with flag "$Flagged"
+    Then the message list is empty
+
+  Scenario: getMessageList filtered by flag should keep flag on non-shared mailbox
+    Given "bob@domain.tld" set flags on "m1" to "$Flagged"
+    When "alice@domain.tld" asks for message list in mailboxes "shared" with flag "$Flagged"
+    Then the message list contains "m1"
+
+  # It does not work for now such as we search by flag first the we filter by inMailbox.
+  # We do not know where can we combine/intersect flag
+  @Ignore
+  Scenario: Get message list should intercept Draft flag on all mailbox
+    Given message "m1" has flags $Draft in mailbox "shared" of user "alice@domain.tld"
+    When "alice@domain.tld" asks for message list in mailboxes "shared,notShared" with flag "$Draft"
+    Then the message list is empty
+
+  Scenario: Get message list should keep flags on non-shared mailbox
+    Given message "m1" has flags $Draft in mailbox "shared" of user "alice@domain.tld"
+    When "alice@domain.tld" asks for message list in mailbox "notShared" with flag "$Draft"
+    Then the message list is empty
+
+  Scenario: Get message list should keep flags on shared mailbox
+    Given message "m1" has flags $Draft in mailbox "shared" of user "alice@domain.tld"
+    When "alice@domain.tld" asks for message list in mailbox "shared" with flag "$Draft"
+    Then the message list contains "m1"
+
+  Scenario: getMessage with shared user should return message with combine flag when delegation mailbox
+    Given message "m1" has flags $Flagged in mailbox "shared" of user "alice@domain.tld"
+    When "bob@domain.tld" ask for messages "m1"
+    Then no error is returned
+    And the keywords of the message is $Flagged
+
+  Scenario: getMessage of owner mailbox should return message with combine flag when delegation mailbox
+    Given message "m1" has flags $Flagged in mailbox "shared" of user "alice@domain.tld"
+    When "alice@domain.tld" ask for messages "m1"
+    Then no error is returned
+    And the keywords of the message is $Flagged
+
+  Scenario: getMessage should keep origin message status when cut the sharing
+    Given message "m1" has flags $Flagged in mailbox "shared" of user "alice@domain.tld"
+    And "alice@domain.tld" shares its mailbox "shared" with rights "" with "bob@domain.tld"
+    When "bob@domain.tld" ask for messages "m1"
+    Then no error is returned
+    And the message has no keywords
```

Comment:
this test makes no sense has bob cannot request m1 anymore, so the result is empty.

Reply:
removed